src/Avalonia.Controls/AppBuilder.cs

```diff
@@ -205,6 +207,27 @@ public AppBuilder With<T>(Func<T> options)
             return Self;
         }
         
+        /// <summary>
+        /// Registers a custom font collection with the font manager.
+        /// </summary>
+        /// <param name="fontCollection">The font collection.</param>
+        /// <returns>An <see cref="AppBuilder"/> instance.</returns>
+        /// <exception cref="ArgumentNullException"></exception>
+        public AppBuilder WithFonts(IFontCollection fontCollection)
+        {
+            if(fontCollection == null)
+            {
+                throw new ArgumentNullException(nameof(fontCollection), "Font collection can't be null.");
+            }
+
+            return AfterSetup(appBuilder =>
+            {
+                var fontManager = FontManager.Current;
+
+                fontManager.AddFontCollection(fontCollection);
+            });
+        }
```

Comment:
I wonder, if we should try to follow MAUI API here, or just do our own.
https://learn.microsoft.com/en-us/dotnet/maui/user-interface/fonts?view=net-maui-7.0#register-fonts

"With***" is more typical to what we usuall do in avalonia.
But it could be:
```csharp
. ConfigureFonts(fonts => // fonts is a FontManager or a wrapper on it with single Add method
{
    fonts.AddFont(new IFontCollection());
    or
    fonts.AddInterFont(); // extension method
})
```
which can be defined in the template like:
```csharp
. ConfigureFonts(fonts => fonts.AddInterFont())
```